cli/providers/aws/route53/cnames.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/pchico83/i2kit/cli/providers/aws"
+	"github.com/pchico83/i2kit/cli/providers/aws/cf"
+	"github.com/pchico83/i2kit/cli/schemas/environment"
+	"github.com/pchico83/i2kit/cli/schemas/service"
+)
+
+//Create creates a CNAME resolving to a service
+func Create(s *service.Service, e *environment.Environment) error {
+	config, err := aws.GetConfig(e.Provider)
+	if err != nil {
+		return err
+	}
+	stack, err := cf.Get(s.Name, config)
+	if err != nil {
+		return err
+	}
+	config, err = aws.GetConfig(e.Root)
+	if err != nil {
+		return err
+	}
+	for _, o := range stack.Outputs {
+		if *o.OutputKey == "elbURL" {
+			svc := route53.New(session.New(), config)
+			hostedZonesInput := &route53.ListHostedZonesByNameInput{
+				DNSName:  awsSDK.String(e.Root.HostedZone),
+				MaxItems: awsSDK.String("1"),
+			}
+			resp, err := svc.ListHostedZonesByName(hostedZonesInput)
+			if err != nil {
+				return err
+			}
+			if len(resp.HostedZones) != 1 {
+				return fmt.Errorf("Hosted zone '%s' not found", e.Root.HostedZone)
+			}
+			hostedZoneID := resp.HostedZones[0].Id
+			recordName := fmt.Sprintf("%s.%s", s.Name, e.Root.HostedZone)
+			params := &route53.ChangeResourceRecordSetsInput{
+				ChangeBatch: &route53.ChangeBatch{
+					Changes: []*route53.Change{
+						{
+							Action: awsSDK.String("UPSERT"),
+							ResourceRecordSet: &route53.ResourceRecordSet{
+								Name: awsSDK.String(recordName),
+								Type: awsSDK.String("CNAME"),
+								ResourceRecords: []*route53.ResourceRecord{
+									{
+										Value: awsSDK.String(*o.OutputValue),
+									},
+								},
+								TTL: awsSDK.Int64(900),
+							},
+						},
+					},
+				},
+				HostedZoneId: hostedZoneID,
+			}
+			_, err = svc.ChangeResourceRecordSets(params)
+			return err
+		}
+	}
+	return fmt.Errorf("No 'elbURL' output found")
+}
+
+//Destroy destroys a CNAME resolving to a service
+func Destroy(s *service.Service, e *environment.Environment) error {
+	config, err := aws.GetConfig(e.Provider)
+	if err != nil {
+		return err
+	}
+	stack, err := cf.Get(s.Name, config)
+	if err != nil {
+		return err
+	}
+	config, err = aws.GetConfig(e.Root)
+	if err != nil {
+		return err
+	}
+	for _, o := range stack.Outputs {
+		if *o.OutputKey == "elbURL" {
+			svc := route53.New(session.New(), config)
+			hostedZonesInput := &route53.ListHostedZonesByNameInput{
+				DNSName:  awsSDK.String(e.Root.HostedZone),
+				MaxItems: awsSDK.String("1"),
+			}
+			resp, err := svc.ListHostedZonesByName(hostedZonesInput)
+			if err != nil {
+				return err
+			}
+			if len(resp.HostedZones) != 1 {
+				return fmt.Errorf("Hosted zone '%s' not found", e.Root.HostedZone)
+			}
+			hostedZoneID := resp.HostedZones[0].Id
+			recordName := fmt.Sprintf("%s.%s", s.Name, e.Root.HostedZone)
+			request := &route53.ChangeResourceRecordSetsInput{
+				ChangeBatch: &route53.ChangeBatch{
+					Changes: []*route53.Change{
+						{
+							Action: awsSDK.String("DELETE"),
```

Comment:
it looks like the entire difference between both functions is this string?> maybe a common func for both?

Reply:
yeap, there are other pending refactors too